From c3df40d5d4ef5768c57d2733203290e27defce08 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Sun, 21 Jul 2019 05:49:44 +0200
Subject: [PATCH 083/106] Revert "fuse: Convert fusectl to use the new mount
 API" (v2)

This reverts commit 29cc02d949b19fdeba9de9f54b2641005f5865c6.

(v2)  Conflict fs/fuse/control.c due to commit
      c23a0bb ("convenience helper: get_tree_single()")
---
 fs/fuse/control.c | 20 +++++---------------
 1 file changed, 5 insertions(+), 15 deletions(-)

diff --git a/fs/fuse/control.c b/fs/fuse/control.c
index cc7e94d73c6c..cdd8f4b60fe8 100644
--- a/fs/fuse/control.c
+++ b/fs/fuse/control.c
@@ -10,7 +10,6 @@
 
 #include <linux/init.h>
 #include <linux/module.h>
-#include <linux/fs_context.h>
 
 #define FUSE_CTL_SUPER_MAGIC 0x65735543
 
@@ -328,7 +327,7 @@ void fuse_ctl_remove_conn(struct fuse_conn *fc)
 	drop_nlink(d_inode(fuse_control_sb->s_root));
 }
 
-static int fuse_ctl_fill_super(struct super_block *sb, struct fs_context *fctx)
+static int fuse_ctl_fill_super(struct super_block *sb, void *data, int silent)
 {
 	static const struct tree_descr empty_descr = {""};
 	struct fuse_conn *fc;
@@ -354,19 +353,10 @@ static int fuse_ctl_fill_super(struct super_block *sb, struct fs_context *fctx)
 	return 0;
 }
 
-static int fuse_ctl_get_tree(struct fs_context *fc)
+static struct dentry *fuse_ctl_mount(struct file_system_type *fs_type,
+			int flags, const char *dev_name, void *raw_data)
 {
-	return get_tree_single(fc, fuse_ctl_fill_super);
-}
-
-static const struct fs_context_operations fuse_ctl_context_ops = {
-	.get_tree	= fuse_ctl_get_tree,
-};
-
-static int fuse_ctl_init_fs_context(struct fs_context *fc)
-{
-	fc->ops = &fuse_ctl_context_ops;
-	return 0;
+	return mount_single(fs_type, flags, raw_data, fuse_ctl_fill_super);
 }
 
 static void fuse_ctl_kill_sb(struct super_block *sb)
@@ -385,7 +375,7 @@ static void fuse_ctl_kill_sb(struct super_block *sb)
 static struct file_system_type fuse_ctl_fs_type = {
 	.owner		= THIS_MODULE,
 	.name		= "fusectl",
-	.init_fs_context = fuse_ctl_init_fs_context,
+	.mount		= fuse_ctl_mount,
 	.kill_sb	= fuse_ctl_kill_sb,
 };
 MODULE_ALIAS_FS("fusectl");
-- 
2.17.1

